fastafetcher.py

```diff
@@ -116,31 +121,27 @@ def main():
         )
 
     # Parse in the multifasta and assign an iterable variable:
-    to_write = []
-    for rec in SeqIO.parse(args.fasta, "fasta"):
-        if args.invert is False:
-            if args.method == "exact":
-                if rec.id in keys:
-                    print(rec.format("fasta"))
-                    to_write.append(rec)
+    # writing directly to outfile, as storing results increases
+    # both runtime and memory footprint
+    with open(args.outfile, 'w') as f:
+        for rec in SeqIO.parse(args.fasta, "fasta"):
+            if args.invert is False:
+                if args.method == "exact":
+                    if rec.id in keys:
+                        f.write(rec.format('fasta'))
+
+                elif args.method == "partial":
+                    if any(key in rec.description for key in keys):
+                        f.write(rec.format('fasta'))
+
+            elif args.invert is True:
+                if args.method == "exact":
+                    if rec.id not in keys:
+                        f.write(rec.format('fasta'))
+                elif args.method == "partial":
+                    if all(key not in rec.description for key in keys):
+                        f.write(rec.format('fasta'))
```

Comment:
Hi, thanks for the suggestions.

I chose to deliberately not require the use of an outfile, or at least preference STDOUT so that the script could be used in unix pipes. This could be worked around by setting the default parameter for `--outfile` to `sys.stdout` potentially...